pyiceberg/table/__init__.py

```diff
@@ -235,6 +242,10 @@ class TableProperties:
     WRITE_PARTITION_SUMMARY_LIMIT = "write.summary.partition-limit"
     WRITE_PARTITION_SUMMARY_LIMIT_DEFAULT = 0
 
+    DELETE_MODE = "write.delete.mode"
+    DELETE_MODE_COPY_ON_WRITE = "merge-on-read"
+    DELETE_MODE_MERGE_ON_READ = "copy-on-write"
```

Comment:
@Fokko I believe this is a mismatch.

Reply:
![oof](https://github.com/apache/iceberg-python/assets/1134248/8ede6002-a16d-4377-91a5-a59bb5c03f48)

That's a big one, thanks @MehulBatra!